x/pocketcore/keeper/session.go

```diff
@@ -55,6 +60,18 @@ func (k Keeper) IsSessionBlock(ctx sdk.Ctx) bool {
 	return ctx.BlockHeight()%k.posKeeper.BlocksPerSession(ctx) == 1
 }
 
+// IsLatestSessionBlockHeightWithinTolerance checks if the latest session block height is bounded by minSessionBlockHeight <= x <= latestSessionBlockHeight
```

Comment:
I have less context about session rollover than node runners, so please correct me if I'm wrong below.

---
```go
// "GetLatestSessionBlockHeight" - Returns the latest session block height (first block of the session, (see blocksPerSession))
```

A session rollover issue happens when:
1. Assume the session starts at `height=96` => `latestSessionHeight=96` for any of `[96, 97, 98, 99]`
2. proofBlockHeight can be in `[96, 97, 98, 99]` to be valid 
_* We encounter issues in this approach in the world today*_
Two things can now happen on a follow-up relay and we want to account for both:
- Dispatch session > node session (e.g. 100 > 96)
- Node session  dispatch session (e.g. 100 > 96(


```go
func (k Keeper) IsProofSessionHeightWithinTolerance(ctx sdk.Ctx, proofSessionHeight int64) bool {
	// Session block height can never be zero.
	if proofSessionHeight <= 0 {
		return false
	}
	sessionTolerance := types.GlobalPocketConfig.ClientSessionSyncAllowance * k.posKeeper.BlocksPerSession(ctx) // num of blocks to tolerate
	blocksPerSession = // get this from global var
	minHeight := proofSessionHeight - sessionTolerance * blocksPerSession
	maxHeight := latestSessionHeight + sessionTolerance * blocksPerSession
	return sdk.IsBetween(proofSessionHeight, minHeight, maxHeight)
}
```

Reply:
Can you define proof block height?  What is proof block height in your example?

Reply:
My initial implementation looked like yours, but ultimately I changed it.

We only want to account for the latter, because if a node height is behind the dispatched one, they will not have the block context to service the relay. I believe this is why I changed it to maxheight to the latest session blockheight that the node is aware of.

Let's just say we do use the max block height tolerance as the maximum range, what if the block height known for that node is `99`? but the actual block height is 100.

Reply:
maxHeight should actually be the node's latest height here for better accuracy of what i was intending, but relays should always include a session block height.

Reply:
Added change for this.

Reply:
> Can you define proof block height? What is proof block height in your example?

Do you mean `proofSessionHeight`? If so, it's the value returned by `GetLatestSessionBlockHeight`

---

Thoughts on my suggestion above re renaming the function to `IsProofSessionHeightWithinTolerance`? I'm having a hard time wrapping my head around `IsLatestSessionHeightWithinTolerance`.

Could you add a godoc comment with an example please?